tests/merch.test.ts

```diff
@@ -0,0 +1,303 @@
+      { uuid: item.uuid },
+      { merchandise: merchItemEdits },
+      admin,
+    );
+
+    const merchItemResponse = await ControllerFactory.merchStore(conn).getOneMerchItem({ uuid: item.uuid }, admin);
+
+    expect(merchItemResponse.item).toStrictEqual({
+      collection: merchItemResponse.item.collection,
+      ...item.getPublicMerchItem(),
+      ...merchItemEdits,
+    });
+  });
+  test('succeeds when basic option fields are updated', async () => {
+    const conn = await DatabaseConnection.get();
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [option1, option2] = MerchFactory.optionsWith(
+      {
+        price: 2500,
+        quantity: 10,
+        discountPercentage: 0,
+        metadata: {
+          type: 'SIZE',
+          value: '12 in.',
+          position: 0,
+        },
+      }, {
+        price: 3500,
+        quantity: 5,
+        discountPercentage: 10,
+        metadata: {
+          type: 'SIZE',
+          value: '24 in.',
+          position: 1,
+        },
+      },
+    );
+    const [item] = MerchFactory.itemsWith({
+      hasVariantsEnabled: true,
+      options: [option1, option2],
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+    });
+
+    await new PortalState()
+      .createUsers([admin])
+      .createMerch([collection])
+      .write();
+
+    option1.price -= 1000;
+    option1.quantity -= 5;
+    option1.discountPercentage = 15;
+    option2.price += 500;
+    option2.quantity += 5;
+    option1.discountPercentage = 0;
+
+    await ControllerFactory.merchStore(conn).editMerchItem(
+      { uuid: item.uuid },
+      {
+        merchandise: {
+          options: [option1, option2],
+        },
+      },
+      admin,
+    );
+
+    const merchItemResponse = await ControllerFactory.merchStore(conn).getOneMerchItem({ uuid: item.uuid }, admin);
+
+    expect(merchItemResponse.item.options).toEqual(
+      expect.arrayContaining([option1.getPublicMerchItemOption(), option2.getPublicMerchItemOption()]),
+    );
+  });
+  test('succeeds when updated item option metadata types are changed but are still consistent', async () => {
+    const conn = await DatabaseConnection.get();
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [metadata1, metadata2, metadata3] = MerchFactory.createOptionMetadata(3);
+    const options = MerchFactory.optionsWith(
+      { metadata: metadata1 },
+      { metadata: metadata2 },
+      { metadata: metadata3 },
+    );
+    const [item] = MerchFactory.itemsWith({
+      hasVariantsEnabled: true,
+      options,
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+    });
+
+    await new PortalState()
+      .createUsers([admin])
+      .createMerch([collection])
+      .write();
+
+    // change every option's type to a different but consistent one
+    options.forEach((option) => { option.metadata.type = 'new type'; });
```

Comment:
Let's do a `map` instead of an assignment in a for-each here.

Reply:
Do you mean something like this?
```ts
options = options.map((option) => ({ ...option, metadata: { type: 'new type' }}));
```
I feel like this is a little less cleaner than using `forEach` since it's a simple assigning vs. reconstructing an object. We'd still need the `options` array for the assertion below so I don't think we can map to anything else.

Reply:
The thing with `forEach` is that it takes a consumer of a single option and does something with the option, but the consumer shouldn't alter the option itself. If we're actually altering options, then we should do it within a `map` and not a `forEach`.